### PR TITLE
Add stub for first

### PR DESCRIPTION
**Pull request by ikonst**
Reviewed by author in https://github.com/hynek/first/pull/22

### CODE REVIEW

File: third_party/2and3/first.pyi [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+@overload
+def first(iterable: Iterable[_T]) -> Optional[_T]: ...
+
+
+@overload
+def first(iterable: Iterable[_T], default: _S) -> Union[_T, _S]: ...
+
+
+@overload
+def first(iterable: Iterable[_T], default: _S, key: Optional[Callable[[_T], Any]]) -> Union[_T, _S]: ...
```
Comment (JelleZijlstra) [inline third_party/2and3/first.pyi]: You're missing support for the case where `key` is given but `default` is not.

Also, don't put any blank lines between the overloads.

**Comment by ikonst** [comment]
> You're missing support for the case where key is given but default is not.

Added. I suppose once dealing with overloads, mypy doesn't take the `default`'s default value of `None` into account.

File: third_party/2and3/first.pyi [hunk 2]
```diff
@@ -5,11 +5,9 @@ _S = TypeVar('_S')
...
-
-
 @overload
 def first(iterable: Iterable[_T], default: _S) -> Union[_T, _S]: ...
-
-
 @overload
 def first(iterable: Iterable[_T], default: _S, key: Optional[Callable[[_T], Any]]) -> Union[_T, _S]: ...
+@overload
+def first(iterable: Iterable[_T], key: Optional[Callable[[_T], Any]]) -> Optional[_T]: ...
```
Comment (JelleZijlstra) [inline third_party/2and3/first.pyi]: You need to mark `key` as keyword-only here (by putting `*` before it).

Comment (ikonst) [reply third_party/2and3/first.pyi]: Ohh, of course!

**Comment by JelleZijlstra** [comment]
That sounds like a mypy bug.

**Pull request merged**